### PR TITLE
Rename Files Workflow Operation Handler

### PR DESCRIPTION
**Pull request by narickmann**
This patch adds a rename file workflow operation, which allows users to rename media files using different metadata.
This can be used to, for example, generate human-readable names for downloads.

### Your pull request should…

* [x] have a concise title
* [x] [close an accompanying issue](https://help.github.com/en/articles/closing-issues-using-keywords) if one exists
* [x] [be against the correct branch](https://docs.opencast.org/develop/developer/development-process#acceptance-criteria-for-patches-in-different-versions)
* [x] include migration scripts and documentation, if appropriate
* [x] pass automated tests
* [ ] have a clean commit history
* [ ] [have proper commit messages (title and body) for all commits](https://medium.com/@steveamaza/e028865e5791)

### CODE REVIEW

File: docs/guides/admin/docs/workflowoperationhandlers/rename-files-woh.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+This can be used to, for example, generate human readable names for downloads.
+
+
+Parameter Table
+---------------
+
+|configuration keys|description                                                                      |
+|------------------|---------------------------------------------------------------------------------|
+|sorce-flavors     |Comma separated list of flavors to select tracks to rename
+|sorce-flavor      |Same as `sorce-flavors` but can only select a single flavor
```
Comment (Arnei) [inline docs/guides/admin/docs/workflowoperationhandlers/rename-files-woh.md]:
```suggestion
|source-flavors     |Comma separated list of flavors to select tracks to rename
|source-flavor      |Same as `source-flavors` but can only select a single flavor
```

**Review by Arnei — APPROVED** [review]
Looks good to me and works in my local OC 10 all-in-one installation.

While not strictly necessary, I'd suggest adding a few "obvious disclaimers" to the WOH readme, such as:
- Make sure that the metadata fields used for renaming contain valid file basenames/extensions, else the operation and thus the entire workflow will fail.
- This WOH only changes the file descriptor, meaning changing the file extension will NOT convert the file to the given file format.
  - Furthermore, changing the file extension may either break subsequent operations or subsequent operations may even revert the change to the file extension.

**Comment by lkiesow** [comment]
> Make sure that the metadata fields used for renaming contain valid file basenames/extensions,

The working file repository will actually sanitize filenames and make them safe. Things like `../` in the title are taken care of.
But you are right that you need a valid file extension. I've added a warning about that in the documentation.

**Comment by lkiesow** [comment]
Also, feel free to squash-merge this pull request.

**Pull request merged**